frontend/packages/console-dynamic-plugin-sdk/src/utils/common.ts

```diff
@@ -33,3 +33,9 @@ export type ExtensionK8sVerb =
   | 'delete'
   | 'deletecollection'
   | 'watch';
+
+// Type for extension hook
+export type ExtensionHook<T, R = any> = (options: R) => ExtensionHookResult<T>;
+
+// Type for extension hook result that returns [data, resolved, error]
+export type ExtensionHookResult<T> = [T, boolean, any];
```

Comment:
I'm not sure if we want to dictate a standard type here. It's really use case dependent.

thoughts @vojtechszocs

Reply:
I had same types being used in catalog extensions. That's why I pulled it into a common type that can be used in both places.